whelktool/scripts/cleanups/2020/11/lxl-3340-add-idkbse-uris.groovy

```diff
@@ -0,0 +1,30 @@
+PrintWriter scheduledForUpdating = getReportWriter("scheduled-updates")
+PrintWriter failedUpdating = getReportWriter("failed-updates")
+
+String where = """
+        collection = 'auth' AND (
+        data#>>'{@graph,1,@type}' = 'Topic' OR
+        data#>>'{@graph,1,@type}' = 'ComplexSubject' OR
+        data#>>'{@graph,1,@type}' = 'Temporal' OR
+        data#>>'{@graph,1,@type}' = 'Geographic' OR
+        data#>>'{@graph,1,@type}' = 'TopicSubdivision')
+        AND (
+        data#>>'{@graph,1,inScheme,@id}' = 'https://id.kb.se/term/barn' OR
+        data#>>'{@graph,1,inScheme,@id}' = 'https://id.kb.se/term/sao')
+        AND NOT (
+        data#>>'{@graph,0,mainEntity}' LIKE '%id.kb.se%')
```

Comment:
Perhaps a nitpick, but isn't there an @id missing here? :)
If the string "id.kb.se" exists somewhere (else) in the mainEntity part of the record, you'll incorrectly exclude that record.

Reply:
Right! Adding the '@id' doesn't make any difference in this particular case though.